martin-mbtiles/src/tile_copier.rs

```diff
@@ -6,37 +6,87 @@ use crate::{MbtError, Mbtiles};
 use sqlx::sqlite::{SqliteArguments, SqliteConnectOptions};
 use sqlx::{query, query_with, Arguments, Connection, Row, SqliteConnection};
 use std::collections::HashSet;
+use std::ffi::OsStr;
 use std::path::PathBuf;
+#[cfg(feature = "cli")]
+use {clap::error::ErrorKind, clap::Args};
 
 #[derive(Clone, Default, Debug)]
+#[cfg_attr(feature = "cli", derive(Args))]
 pub struct TileCopierOptions {
-    zooms: HashSet<u8>,
+    /// MBTiles file to read from
+    src_file: PathBuf,
+    /// MBTiles file to write to
+    dst_file: PathBuf,
+    /// Force the output file to be in a simple MBTiles format with a `tiles` table
+    #[cfg_attr(feature = "cli", arg(long))]
+    force_simple: bool,
+    /// Minimum zoom level to copy
+    #[cfg_attr(feature = "cli", arg(long))]
     min_zoom: Option<u8>,
+    /// Maximum zoom level to copy
+    #[cfg_attr(feature = "cli", arg(long))]
     max_zoom: Option<u8>,
-    //self.bbox = bbox
-    verbose: bool,
+    /// List of zoom levels to copy; if provided, min-zoom and max-zoom will be ignored
+    #[cfg_attr(feature = "cli", arg(long, value_parser(clap::builder::ValueParser::new(HashSetValueParser{}))))]
+    zoom_levels: HashSet<u8>,
+}
+
+#[cfg(feature = "cli")]
+#[derive(Clone)]
+struct HashSetValueParser;
+
+#[cfg(feature = "cli")]
+impl clap::builder::TypedValueParser for HashSetValueParser {
+    type Value = HashSet<u8>;
+
+    fn parse_ref(
+        &self,
+        _cmd: &clap::Command,
+        _arg: Option<&clap::Arg>,
+        value: &OsStr,
+    ) -> Result<Self::Value, clap::Error> {
+        let mut result = HashSet::<u8>::new();
+        let values = value
+            .to_str()
+            .ok_or(clap::Error::new(ErrorKind::ValueValidation))?
+            .split(',');
+        for val in values {
+            result.insert(
+                val.parse::<u8>()
+                    .map_err(|_| clap::Error::new(ErrorKind::ValueValidation))?,
+            );
+        }
+        Ok(result)
+    }
 }
 
 #[derive(Clone, Debug)]
 pub struct TileCopier {
     src_mbtiles: Mbtiles,
-    dst_filepath: PathBuf,
     options: TileCopierOptions,
 }
 
 impl TileCopierOptions {
-    pub fn new() -> Self {
+    pub fn new(src_filepath: PathBuf, dst_filepath: PathBuf) -> Self {
         Self {
-            zooms: HashSet::new(),
+            src_file: src_filepath,
+            zoom_levels: HashSet::new(),
+            force_simple: false,
             min_zoom: None,
             max_zoom: None,
-            verbose: false,
+            dst_file: dst_filepath,
         }
     }
 
-    pub fn zooms(mut self, zooms: Vec<u8>) -> Self {
-        for zoom in zooms {
-            self.zooms.insert(zoom);
+    pub fn force_simple(mut self, force_simple: bool) -> Self {
+        self.force_simple = force_simple;
+        self
+    }
+
+    pub fn zoom_levels(mut self, zoom_levels: Vec<u8>) -> Self {
+        for zoom in zoom_levels {
```

Comment:
you might be able to use `.extend(...)` method, but not certain

Reply:
I did. FYI I could also do that here: https://github.com/maplibre/martin/pull/723/files#diff-df5958e4bb210418bbc3f1ac707f8df93314ad138dfa40d911eb8b0e51b1fc8cR54-R59 but I avoided it because the necessary mapping makes it much less less readable